src/libcollections/binary_heap.rs

```diff
@@ -568,6 +568,9 @@ pub struct Iter <'a, T: 'a> {
     iter: slice::Iter<'a, T>,
 }
 
+unsafe impl<'a, T: Sync> Sync for Iter<'a, T> {}
+unsafe impl<'a, T: Sync> Send for Iter<'a, T> {}
```

Comment:
Shouldn't these be unnecessary? I would figure that `slice::Iter` would implement these as appropriate